### PR TITLE
pkg: add missing xfrm-no-track rules from ipv6

### LINKED ISSUE #23481: ipsec: IPv6 externalTrafficPolicy=Local E/W is broken when accessed from non-backend node
The [test case](https://github.com/cilium/cilium/blob/v1.13.0-rc5/test/k8s/service_helpers.go#L868) is broken for IPv6 when running with IPsec with vxlan and KPR=disabled. However, it works for IPv4 service.

The tcpdump on a node (`fc00:f853:ccd:e793::4`) which issued request:
```
09:08:18.623453 cilium_vxlan P   IP6 fc00:f853:ccd:e793::4.50304 > fd02::11a.80: Flags [S], seq 4061851243, win 65476, options [mss 65476,sackOK,TS val 3818350863 ecr 0,nop,wscale 7], length 0
09:08:18.623588 lxc9aafe1894701 Out IP6 fc00:f853:ccd:e793::4.50304 > fd02::11a.80: Flags [S], seq 4061851243, win 65476, options [mss 65476,sackOK,TS val 3818350863 ecr 0,nop,wscale 7], length 0
09:08:18.624153 lxc9aafe1894701 In  IP6 fd02::11a.80 > fc00:f853:ccd:e793::4.50304: Flags [S.], seq 108698648, ack 4061851244, win 65050, options [mss 1313,sackOK,TS val 2604710198 ecr 3818350863,nop,wscale 7], length 0
09:08:18.624845 eth0  Out IP6 fd02::11a.80 > fc00:f853:ccd:e793::4.50304: Flags [S.], seq 108698648, ack 4061851244, win 65050, options [mss 1313,sackOK,TS val 2604710198 ecr 3818350863,nop,wscale 7], length 0
09:08:18.627947 cilium_vxlan In  IP6 fc00:f853:ccd:e793::4.50304 > fd02::11a.80: Flags [R], seq 4061851244, win 0, length 0
```

The service was accessed via `curl 'http://[fc00:f853:ccd:e793::4]:31088`, and the backend is running on a `fc00:f853:ccd:e793::5` node.

Spot that the reply is coming via `eth0`, so it might be another instance of https://github.com/cilium/cilium/issues/23461.

The relevant pwru output:
```
0xffff8ad95a6c1b00      4    [ksoftirqd/4]         ip6_input_finish sec-arg=0 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=80 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] ip6_protocol_deliver_rcu sec-arg=18446615264862411520 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=80 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]       raw6_local_deliver sec-arg=6 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]               tcp_v6_rcv sec-arg=96 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]    inet6_lookup_listener sec-arg=18446744071712769600 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]      inet6_lhash2_lookup sec-arg=18446615260171014720 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]      inet6_lhash2_lookup sec-arg=18446615260171015856 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]      __xfrm_policy_check sec-arg=0 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]          decode_session6 sec-arg=18446633258186882072 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] security_xfrm_decode_session sec-arg=18446633258186882092 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] bpf_lsm_xfrm_decode_session sec-arg=18446633258186882092 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] __nf_nat_decode_session      [nf_nat] sec-arg=18446633258186882072 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]           tcp_v6_fill_cb sec-arg=18446615264978099440 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]        tcp_v6_send_reset sec-arg=18446615264862411520 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]     tcp_v6_send_response sec-arg=18446615264862411520 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] security_skb_classify_flow sec-arg=18446633258186881904 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4] bpf_lsm_xfrm_decode_session sec-arg=18446633258186881924 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
0xffff8ad95a6c1b00      4    [ksoftirqd/4]         kfree_skb_reason sec-arg=2 netns=4026534705 mark=0x0 ifindex=32 proto=dd86 mtu=1500 len=40 [fd02::11a]:80->[fc00:f853:ccd:e793::4]:41453(tcp)
```

### PR DESCRIPTION
**Pull request by jschwinger233**
By right there should be a rule to let ipsec skb bypass conntrack:

-A CILIUM_PRE_raw -m mark --mark 0xd00/0xf00 -m comment --comment "cilium-xfrm-notrack:" -j CT --notrack

However ipv6 missed it and this commit adds the rule back.

Fixes: #23481

```release-note
Add missing xfrm-no-track rules for IPv6 IPSec. This fixes a connectivity issue for IPv6 IPSec with externalTrafficPolicy=local.
```

### CODE REVIEW

File: pkg/datapath/iptables/iptables.go [hunk 1]
```diff
@@ -1640,9 +1640,14 @@ func (m *IptablesManager) addCiliumAcceptXfrmRules() error {
...
-func (m *IptablesManager) addCiliumNoTrackXfrmRules() error {
+func (m *IptablesManager) addCiliumNoTrackXfrmRules() (err error) {
 	if option.Config.EnableIPv4 {
-		return m.ciliumNoTrackXfrmRules(ip4tables, "-I")
+		if err = m.ciliumNoTrackXfrmRules(ip4tables, "-I"); err != nil {
+			return
+		}
+	}
+	if option.Config.EnableIPv6 {
+		return m.ciliumNoTrackXfrmRules(ip6tables, "-I")
```
Comment (pchaigno) [inline pkg/datapath/iptables/iptables.go:1650]: Do we know if this issue was always there or if it's a regression? (Probably doesn't matter if it was introduced before v1.11.0.)

**Review by pchaigno — COMMENTED** [review]
Nice! Thanks for tracking and fixing this!

I see it's labeled as `release-note/minor`, but it feels more like a `release-note/bug`, no? The release note should also describe the user impact. It currently describes the change itself.
Finally, I think we should figure out where we need to backport.

File: pkg/datapath/iptables/iptables.go (hunk 1 above)
Comment (jschwinger233) [reply pkg/datapath/iptables/iptables.go:1650]: v1.11.0 has the exactly same code. 

The code history indicates is was always there.
image

**Comment by jschwinger233** [comment]
> I think we should figure out where we need to backport.

I noticed there is no PR labelled with `backport/1.10` and earlier versions, so I add `backport/1.11` for this, if it's not appropriate please let me know.

**Review by brb — APPROVED** [review]
:rocket:

**Comment by julianwiedmann** [comment]
[word-smith'ed the release-note. Also adding backport labels for 1.12 and 1.13]

File: pkg/datapath/iptables/iptables.go (hunk 1 above)
Comment (pchaigno) [reply pkg/datapath/iptables/iptables.go:1650]: Thanks for checking!

**Review by pchaigno — APPROVED** [review]

**Comment by jschwinger233** [comment]
Backport PR will be taken care of by renovate bot, right? Is there anything I should follow up?

**Comment by pchaigno** [comment]
> Backport PR will be taken care of by renovate bot, right? Is there anything I should follow up?

Backport PRs will be opened by the backporter. If that person can't backport themselves because it's too involved, they will ping you, in the backport PR or here.

**Pull request merged**